octopusdeploy3-single-vm-windows/azuredeploy.json

```diff
@@ -0,0 +1,355 @@
+        "North Europe",
+        "South Central US",
+        "Southeast Asia",
+        "West Europe",
+        "West US"
+      ],
+      "metadata": {
+        "Description": "Location of the Octopus Deploy server."
+      }
+    },
+    "vmStorageAccountName": {
+      "type": "string",      
+      "metadata": {
+        "description": "Unique DNS Name for the Storage Account that will hold the Octopus Deploy Virtual Machine Disks."
+      }
+    },
+    "vmAdminUsername": {
+      "type": "string",
+      "defaultValue": "octoadmin",
+      "metadata": {
+        "description": "Admin username for the Octopus Deploy Virtual Machine."
+      }
+    },
+    "vmAdminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Admin password for the Octopus Deploy Virtual Machine."
+      }
+    },
+    "networkDnsName": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Name used to access the Octopus Deploy server via HTTP or RDP."
+      }
+    },
+    "sqlServerName": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Name for the SQL DB Server that will hold the Octopus Deploy data."
+      }
+    },
+    "sqlAdminUsername": {
+      "type": "string",
+      "defaultValue": "sqladmin",
+      "metadata": {
+        "description": "Admin username for the Octopus Deploy SQL DB Server."
+      }
+    },
+    "sqlAdminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Admin password for the Octopus Deploy SQL DB Server."
+      }
+    },
+    "licenseFullName": { 
+      "type": "string",
+      "metadata": {
+        "description": "Octopus Deploy Trial license - provide Full Name."
+      }
+    },
+    "licenseOrganisationName": { 
+      "type": "string",
+      "metadata": {
+        "description": "Octopus Deploy Trial license - provide Organisation Name."
+      }
+    },
+    "licenseEmailAddress": { 
+      "type": "string",
+      "metadata": {
+        "description": "Octopus Deploy Trial license - provide Email Addres."
+      }
+    },
+    "octopusAdminUsername" : {
+      "type": "string",
+      "defaultValue": "admin",
+      "metadata": {
+        "description": "Admin username for the Octopus Deploy web application."
+      }
+    },
+    "octopusAdminPassword" : {
+      "type": "securestring",
+      "metadata": {
+        "description": "Admin password for the Octopus Deploy web application."
+      }
+    }
+  },
+  "variables": {
+    "location": "[parameters('location')]",
+    "vmImagePublisher": "MicrosoftWindowsServer",
+    "vmImageOffer": "WindowsServer",
+    "vmOSDiskName": "osdiskforwindowssimple",
+    "vmWindowsOSVersion": "2012-R2-Datacenter",
+    "vmStorageAccountName": "[parameters('vmStorageAccountName')]",
+    "vmStorageAccountType": "Standard_LRS",
+    "vmStorageAccountContainerName": "vhds",
+    "vmName": "OctopusDeploy",
+    "vmSize": "Standard_D2",
```

Comment:
You can have a list of allowed sizes that the user can select from so that the exact size string is visible for the user. In Portal deploy experience, this would be a drop down box.